test/boost/frozen_mutation_test.cc

```diff
@@ -169,3 +170,19 @@ SEASTAR_TEST_CASE(test_deserialization_using_wrong_schema_throws) {
         }
     });
 }
+
+SEASTAR_TEST_CASE(frozen_mutation_is_consumed_in_order) {
+    random_mutation_generator gen{random_mutation_generator::generate_counters::no};
+    mutation m = gen();
+    auto& s = m.schema();
+    frozen_mutation fm{m};
```

Comment:
Please also test with `mutation`s resulting from adding several different mutations together.

Reply:
What do you mean by "adding" mutation together?

Reply:
```
auto m1 = gen();
auto m2 = gen();
auto m = m1 + m2;
```
I don't think this will find anything but just to be sure...

Reply:
Done in v2